ios/Classes/core/PMManager.m

```diff
@@ -440,15 +440,22 @@ - (void)fetchThumb:(PHAsset *)asset option:(PMThumbLoadOption *)option resultHan
                       contentMode:option.contentMode
                           options:requestOptions
                     resultHandler:^(PMImage *result, NSDictionary *info) {
-        BOOL downloadFinished = [PMManager isDownloadFinish:info];
+        if ([handler isReplied]) {
+            return;
+        }
 
-        if (!downloadFinished) {
+        if (info[PHImageErrorKey]) {
+            [handler reply: nil];
+            [self notifySuccess:progressHandler];
             return;
         }
 
-        if ([handler isReplied]) {
+        BOOL downloadFinished = [PMManager isDownloadFinish:info];
+
+        if (!downloadFinished) {
```

Comment:
Can we inline this?

Reply:
Do you mean like `if (![PMManager isDownloadFinish:info]) {`? I can do that but it will be inconsistent with other parts of your code.